#160056181 User Login, Password Reset Hosting the Frontend on Heroku

yomigeek opened the pull request:
#### What does this PR do?
- Add ability for users to log in and reset their password and host the application on Heroku
#### Description of Task to be completed?
- add user login feature
- add user reset password feature
- fix some CSS mix-ups
- add API proxy link for API calls in development
- implement feedback from the demo
- move query-sting module to /src
- host the app on Heroku
- add static.json to fix react router error on Heroku
#### How should this be manually tested?
- Clone the repository 
- Navigate into the projects directory using `cd elven-ah-client`
- Run `npm install`
- Run `npm start`
- Navigate to /login and /password/reset and try it out 

#### What are the relevant pivotal tracker stories?
[#160056181](https://www.pivotaltracker.com/story/show/160056181)
[#160056181](https://www.pivotaltracker.com/story/show/160056180)
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-17 at 2 47 44 pm" src="https://user-images.githubusercontent.com/26969309/45627368-ba14a080-ba89-11e8-9a8a-737130c74c96.png">
<img width="1440" alt="screen shot 2018-09-17 at 2 47 53 pm" src="https://user-images.githubusercontent.com/26969309/45627369-ba14a080-ba89-11e8-8860-62a678eff864.png">
<img width="1440" alt="screen shot 2018-09-17 at 2 48 21 pm" src="https://user-images.githubusercontent.com/26969309/45627371-ba14a080-ba89-11e8-9787-6cc1c3441506.png">

#### Questions:
- None

madeofhuman committed:
fix(ui): fix background image scaling (#12)

- fix background image scaling
[Delivers #160514742]

Veraclins committed:
#160056181 User Login, Password Reset Hosting the Frontend on Heroku (#13)

* login and password

* feat(auth): user login, reset password and others

- add user login feature
- add user reset password feature
- fix some css mix ups
- add API proxy link for API calls in development
- implement feedback from demo
- move query-sting module to /src
- add static.json to fix react router error on heroku

[Delivers #160056181 #160056180]

* remove folder for current task

* add global history object

tonymontaro commented on src/features/authentication/password-reset/passwordResetAction.js:
```diff
@@ -9,41 +9,51 @@ import {
+  }
+  if (response.status >= 500) {
+    dispatch({ type: NETWORK_ERROR });
+    return toastr.error('Oops! unable to connect to the Internet. Please check your connection and try again');
```
I see this line repeated several times, let's try to make the codebase DRY.

tonymontaro reviewed (changes requested):
Good job! Just one minor issue to be addressed.

Commit:
feat(auth): fix logout issues and refactor

- remove user initialization that keeps user object in state after logout
- refactor error handling
